tt_metal/impl/dispatch/kernels/remote_issue_queue_interface.cpp

```diff
@@ -0,0 +1,95 @@
+        uint32_t data_size = command_ptr[DeviceCommand::data_size_idx];
+        uint32_t num_buffer_transfers = command_ptr[DeviceCommand::num_buffer_transfers_idx];
+        uint32_t page_size = command_ptr[DeviceCommand::page_size_idx];
+        uint32_t producer_cb_size = command_ptr[DeviceCommand::producer_cb_size_idx];
+        uint32_t consumer_cb_size = command_ptr[DeviceCommand::consumer_cb_size_idx];
+        uint32_t producer_cb_num_pages = command_ptr[DeviceCommand::producer_cb_num_pages_idx];
+        uint32_t consumer_cb_num_pages = command_ptr[DeviceCommand::consumer_cb_num_pages_idx];
+        uint32_t num_pages = command_ptr[DeviceCommand::num_pages_idx];
+        uint32_t producer_consumer_transfer_num_pages = command_ptr[DeviceCommand::producer_consumer_transfer_num_pages_idx];
+        uint32_t sharded_buffer_num_cores = command_ptr[DeviceCommand::sharded_buffer_num_cores_idx];
+        uint32_t wrap = command_ptr[DeviceCommand::wrap_idx];
+
+        if ((DeviceCommand::WrapRegion)wrap == DeviceCommand::WrapRegion::ISSUE) {
+            // Basically popfront without the extra conditional
+            cq_read_interface.issue_fifo_rd_ptr = cq_read_interface.issue_fifo_limit - cq_read_interface.issue_fifo_size;  // Head to beginning of command queue
+            cq_read_interface.issue_fifo_rd_toggle = not cq_read_interface.issue_fifo_rd_toggle;
+            notify_host_of_issue_queue_read_pointer();
+            continue;
+        }
+
+        program_local_cb(data_section_addr, producer_cb_num_pages, page_size, producer_cb_size);
+        while (db_semaphore_addr[0] == 0)
+            ;  // Check that there is space in the consumer
+        // program_consumer_cb(db_buf_switch, ((uint64_t)consumer_noc_encoding << 32), consumer_cb_num_pages, page_size, consumer_cb_size);
+        // relay_command(db_buf_switch, ((uint64_t)consumer_noc_encoding << 32));
+
+        // Decrement the semaphore value
+        noc_semaphore_inc(((uint64_t)producer_noc_encoding << 32) | uint32_t(db_semaphore_addr), -1);  // Two's complement addition
+        noc_async_write_barrier();
+
+        // Notify the consumer
+        // noc_semaphore_inc( ((uint64_t)consumer_noc_encoding << 32) | get_semaphore(0), 1);
+        // noc_async_write_barrier();  // Barrier for now
+
+        // Fetch data and send to the consumer
+        // produce(
+        //     command_ptr,
+        //     num_buffer_transfers,
+        //     sharded_buffer_num_cores,
+        //     page_size,
+        //     producer_cb_size,
+        //     producer_cb_num_pages,
+        //     consumer_cb_size,
+        //     consumer_cb_num_pages,
+        //     ((uint64_t)consumer_noc_encoding << 32),
+        //     producer_consumer_transfer_num_pages,
+        //     db_buf_switch);
+
+        issue_queue_pop_front(DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND + data_size);
+
+        // db_buf_switch = not db_buf_switch; // only 1 command slot on consumer side
```

Comment:
Why all the commented out pieces of code?

Reply:
commented out code will be updated in Allans PR for FD tunnelling because eth core syncs with this core